src/apps/chatter/hooks/use-room.js

```diff
@@ -3,23 +3,28 @@ import { useRef, useEffect, useState } from 'react'
 import url from '/util/url'
 import { getState } from '/store'
 
-export function useRoom (roomId) {
+export function useRoom (roomId, password) {
   const [entries, setEntries] = useState([])
+  const [isAuthorized, setIsAuthorized] = useState(false)
+  const [isInitialized, setIsInitialized] = useState(false)
   const socketRef = useRef()
 
   const NEW_ENTRY = 'NEW_ENTRY'
   const INIT_CHAT = 'INIT_CHAT'
   const FETCH_ENTRIES = 'FETCH_ENTRIES'
   const ENTRIES = 'ENTRIES'
+  const INIT_RESPONSE = 'INIT_RESPONSE'
 
   useEffect(() => {
+    console.log('USING EFFECT SUBJECT TO PASSWORD: ', password)
     connect()
     return () => {
       socketRef.current.close()
     }
-  }, [])
+  }, [password])
```

Comment:
This would make sense if you were referencing `password` inside `connect` but it doesn't look like you are?

Reply:
it's being used inside initChatUser, which is inside connect. Without that, the init function wont run again to get authorization again on password protected rooms.

Reply:
Gotcha.

```
      initChatUser()
      sendFetchCommand()
```
Those being called at the sametime, does that leave room for a race condition? Does fetching messages expect the `initChatUser` to have run on the server already?

Can we simplify this somehow? What does the initChatUser command accomplish?  Looking at the API code, I'm struggling to see what it does? It responds if the password is valid.. but each subsequent message checks the password as well, no?

Could we remove a step and just try to fetch messages? If the password is wrong, it will return an error message, no?

Reply:
It is indeed a bit annoying that password protected rooms incur a failed fetch on each load. I'll try and think of a way to eliminate the initialization step altogether, along with the associated state boolean